File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    let posArr = [], negArr = [];
    nums.forEach((num) => {
        if(num>=0)
            posArr.push(num);
        else
            negArr.push(num);
    })
    for(let i=0; i<nums.length/2; i++){
        nums[i*2]   = posArr[i]
        nums[i*2+1] = negArr[i]
    }
    return nums;
};

File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] res = new int[nums.length];
        int pos=0, neg=1;

        for(int i=0; i<nums.length; i++) {
            if(nums[i] >=0 ){
                res[pos] = nums[i];
                pos += 2;
            } else {
                res[neg] = nums[i];
                neg += 2;      
            }
        }

        return res;
    }
}